kubernetes/samples/scripts/create-weblogic-domain/model-in-image/cm-docker-file/Dockerfile

```diff
@@ -0,0 +1,40 @@
+# Copyright (c) 2021, Oracle and/or its affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+
+# This is a sample Dockerfile for supplying Model in Image model files
+# and a WDT install in a small separate "common mount" 
+# image. This is an alternative to supplying the files directly
+# in the domain resource `domain.spec.image` image.
+
+# COMMON_MOUNT_PATH arg:
+#   Parent location for Model in Image model and WDT installation files.
+#   Must match domain resource 'domain.spec.commonMountVolumes.mountPath'
+#   Default '/common'.
+#
+# WDT_MODEL_HOME arg:
+#   Location for WebLogic Deploy Tooling model home.
+#   Must match domain resource 'domain.spec.configuration.model.modelHome'
+#   Must be a directory within COMMON_MOUNT_PATH.
+#   Defaults to '${COMMON_MOUNT_PATH}/models'.
+#
+# WDT_INSTALL_HOME arg: 
+#   Location of the WebLogic Deploy Tooling installation.
+#   Must match domain resource 'domain.spec.configuration.model.wdtInstallHome'
+#   Must be a directory within COMMON_MOUNT_PATH.
+#   Defaults to '${COMMON_MOUNT_PATH}/weblogic-deploy'.
+
+FROM busybox
+ARG COMMON_MOUNT_PATH=/common
+ARG WDT_MODEL_HOME=${COMMON_MOUNT_PATH}/models
+ARG WDT_INSTALL_HOME=${COMMON_MOUNT_PATH}/weblogic-deploy
+ARG USER=oracle
+ARG USERID=1000
+ARG GROUP=root
+ENV COMMON_MOUNT_PATH=${COMMON_MOUNT_PATH}
+ENV WDT_MODEL_HOME=${WDT_MODEL_HOME}
+ENV WDT_INSTALL_HOME=${WDT_INSTALL_HOME}
+        RUN adduser -D -u ${USERID} -G $GROUP $USER
+        COPY weblogic-deploy/ ${WDT_INSTALL_HOME}/
+        COPY models/ ${WDT_MODEL_HOME}/
+        RUN chown -R $USER:$GROUP ${COMMON_MOUNT_PATH}/
```

Comment:
You can do the chown as part of the COPY command rather than as a separate RUN... https://docs.docker.com/engine/reference/builder/#copy

Reply:
It's probably more efficient to chown the top-level COMMON_MOUNT_PATH in the same RUN where you add the user and then chown on the two copies... Doing a chown after the copy doubles the size of those files in the image.

Reply:
I tried `chown` on the copies using USER and GROUP variables (like below) but it fails with the following error -
COPY --chown=${USER}:${GROUP} weblogic-deploy/ ${WDT_INSTALL_HOME}/

```
Step 12/14 : COPY --chown=${USER}:${GROUP} weblogic-deploy/ ${WDT_INSTALL_HOME}/
unable to convert uid/gid chown string to host mapping: can't find uid for user ${USER}: no such user: ${USER}
```
This works if I hard code the user and group during COPY -
```
COPY --chown=oracle:root weblogic-deploy/ ${WDT_INSTALL_HOME}/
```
This issue is reported in https://github.com/moby/moby/issues/37290 (which is closed) but I still see it in docker version `19.03.11-ol, build 2a9c376`. For now, I left it as it is, please let me know if I missed something. Thanks.